### ISSUE TITLE
Consider wrapping existing embedded graph db

### ISSUE THREAD

**Issue by stevesimmons** [pmikstacki/SliccDB#14]
Hi,

I came to this project when looking for an embedded graph database for C#. 

As you are at a very early stage, is it ok to ask why are you implementing the whole graph db yourself rather than using an existing embedded graph db engine? 

One other project I follow is Kuzu <https://github.com/kuzudb/kuzu/>. It might be easier/faster to take Kuzu's C++ engine, which runs OpenCypher, and write a C# wrapper.

**Comment by pmikstacki** [comment]
Hi there! 
Thank you for interest in this project.
I had a look at kuzu, and it indeed seems like a feature complete solution. 
SliccDB is my pet project. I used it as a project for my CS class, but then it evolved into something more. 
To answer your question in a more clean fashion I will split my answer to your question in sections:

## The abandonment of OpenCypher in SliccDB codebase
I considered, and even implemented some features of OpenCypher and then pulled out of it completely, mainly because I got familiar with Apache Gremlin and came out with my own solution - Fluent API for graph databases.
That's why I wrote extension methods on the connection. It just seems more logical in the context of c# as a language.

## Performance
SliccDB uses message pack as its core, which makes serialization pretty fast. 
And if you need it to be even faster, you can compile it to native code through [Bflat](https://github.com/bflattened/bflat), and it should be as fast as c++ solution.

## TLDR
I do not plan on just using an existing graph db engine. It is just more fun to make something like this on my own. OpenCypher has been replaced by fluent API calls to dbConnection to make it easier in the context of c#
